### PR TITLE
Update reboot_hook to work on 11.4 and Friends List version

### PR DESCRIPTION
**Pull request by GrayJack**
Thanks to TuxSH and your implementation of the reboot patch being super similar.

I tested, and it works!!!

### CODE REVIEW

File: bits/reboot_hook.s [hunk 1]
```diff
@@ -83,6 +84,9 @@ koffset_base:    .word kernel_code-jump_to_kernel-12
 .pool
 
 kernel_code:
+
+    ldr sp, =0x080FF000
+
```
Comment (chaoskagami) [inline bits/reboot_hook.s]: This line shouldn't be needed.

Immediately after this, we jump to the payload, and sp is overwritten before any stack operations are performed.

Comment (GrayJack) [reply bits/reboot_hook.s]: Looks like it works without that line, but I only tested GBA games cause my TWL_FIRM or corbenik or my A9LH are broken and don't let corbenik load and patch TWL_FIRM... I'll wait for Crystal The Glacion confirm if it works on TWL games/homebrews.
I'll make the changes after that ass soon as possible

**Comment by GrayJack** [comment]
Just did it, and thanks for help me to get better at asm, I'm pretty new at this

**Comment by chaoskagami** [comment]
Nah, it's a tiny unnecessary thing.

TBH, I don't think Luma needs that line either, which makes it kind of strange.

**Comment by GrayJack** [comment]
I think Luma needs that line, I tried remove it and launch Luma and it gets a prefetch abort (svcBreak)

**Comment by chaoskagami** [comment]
It probably has to do with their exception handler needing a valid stack, then.

**Pull request merged**